### PR TITLE
Change logic on token via env variable not set

### PR DESCRIPTION
**Pull request by jeanCarloMachado**
- Add warning when DATABRICKS_TOKEN is set rather than failing when its not set. The bulk of our use-cases rely on the token being set via databricks configure command. The token via environment variable is only used for CI and we should treat as an edge case.

### CODE REVIEW

**Review by Dharin-shah — APPROVED** [review]

File: tests/resources/poetry-test/pyproject.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "poetry-test"
-version = "0.1.0"
+version = "0.1.1"
```
Comment (oliviagyg) [inline tests/resources/poetry-test/pyproject.toml]: What is this?

File: rocket/rocket.py [hunk 2]
```diff
@@ -73,8 +73,8 @@ def launch(
         if not os.path.exists(f"{home}/.databrickscfg"):
             raise Exception("Databricks cli not configured. Run `databricks configure --token`.")
 
-        if os.getenv("DATABRICKS_TOKEN") is None:
-            raise Exception("DATABRICKS_TOKEN must be set for db-rocket to work")
+        if os.getenv("DATABRICKS_TOKEN"):
+            print("Note: DATABRICKS_TOKEN is set, it could override the token in ~/.databrickscfg and cause errors.")
```
Comment (oliviagyg) [inline rocket/rocket.py:77]: I actually do set `DATABRICKS_TOKEN` in my environment rather than use `databricks configure` -- I wonder how common this is. I don't think it changes much either way though.

**Review by oliviagyg — APPROVED** [review]
See comments, otherwise lgtm

File: rocket/rocket.py (hunk 2 above)
Comment (jeanCarloMachado) [reply rocket/rocket.py:77]: I think we should recommend 1 way, so as to keep it simple as one can affect the other and cause debugging problems.  I see developers putting secrets on startup files and I think ideally we desincourage that direction as leaves the laptop leaks more serious. So my opinion is to leave this to be managed by databricks (even though actually its just stored plain somewhere else atm).  Ideally databricks should provide 2factor authentication, or GYG.

File: tests/resources/poetry-test/pyproject.toml (hunk 1 above)
Comment (jeanCarloMachado) [reply tests/resources/poetry-test/pyproject.toml]: Thanks for catching! :)
